pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -773,3 +773,43 @@ def test_slice_bounds_empty(self):
         left = empty_idx._maybe_cast_slice_bound('2015-01-02', 'left', 'loc')
         exp = Timestamp('2015-01-02 00:00:00')
         self.assertEqual(left, exp)
+
+    def test_to_native_types(self):
+        index = DatetimeIndex(freq='1D', periods=3, start='2017-01-01')
+
+        # First, with no arguments.
+        expected = np.array(['2017-01-01', '2017-01-02',
+                             '2017-01-03'], dtype=object)
+
+        result = index.to_native_types()
+        tm.assert_numpy_array_equal(result, expected)
+
+        # No NaN values, so na_rep has no effect
+        result = index.to_native_types(na_rep='pandas')
+        tm.assert_numpy_array_equal(result, expected)
+
+        # Make sure slicing works
+        expected = np.array(['2017-01-01', '2017-01-03'], dtype=object)
+
+        result = index.to_native_types([0, 2])
+        tm.assert_numpy_array_equal(result, expected)
+
+        # Make sure date formatting works
+        expected = np.array(['01-2017-01', '01-2017-02',
+                             '01-2017-03'], dtype=object)
+
+        result = index.to_native_types(date_format='%m-%Y-%d')
```

Comment:
hmm, can you move these to ``test_formats.py`` (new file), in this directory, same with the period ones. I don't these are actually tested anywhere :>